src/pages/accountSetup/AccountSetupPage.jsx

```diff
@@ -0,0 +1,378 @@
+import { updateDoc } from "firebase/firestore";
+import CustomContainer from "../../components/container/CustomContainer";
+import CustomSelect from "../../components/customSelect/CustomSelect";
+import { Typography } from "@mui/material";
+import Grid from "@mui/material/Grid";
+import CustomTextField from "../../components/textField/CustomTextField";
+
+// Validation schema, update where necessary.
+const accountSetupSchema = yup.object({
+  headingsFont: yup.string().required("Please select a heading font"),
+  headingPointSize: yup
+    .number()
+    .required("Please select a heading point size")
+    .min(10, "Please select a heading point size from 10 to 30")
+    .max(25, "Please select a heading point size from 10 to 30"),
+  mainTextFont: yup.string().required("Please select a main text font"),
+  mainTextPointSize: yup
+    .number()
+    .required("Please select a main text point size")
+    .min(10, "Please select a main text point size from 10 to 30")
+    .max(25, "Please select a main text point size from 10 to 30"),
+  systemName: yup.string().required("Please enter a system name"),
+});
+
+const StyledCheckbox = styled(Checkbox)(({ theme }) => ({
+  color: `white`,
+  "&.Mui-checked": {
+    color: `white`,
+  },
+  "& .MuiSvgIcon-root": {
+    fontSize: "2rem",
+  },
+}));
+
+const StyledFormHelperText = styled(FormHelperText)(({ theme }) => ({
+  color: `${theme.palette.error.main}`,
+  fontFamily: `${theme.typography.body1.fontFamily}`,
+  fontSize: "1.2rem",
+}));
+
+const StyledContainer = styled(Box)(({ theme }) => ({
+  backgroundColor: theme.palette.primary.main,
+  borderWidth: 3,
+  borderStyle: "solid",
+  borderColor: theme.palette.secondary.main,
+  boxShadow: `5px 5px 3px -1px #131415`, //${theme.palette.primary.main}`,
+  borderRadius: 12,
+  padding: 0,
+  maxWidth: "none",
+
+  justifyContent: "center",
+  alignItems: "center",
+  color: theme.palette.text.primary,
+  height: "95%",
+  width: "98%",
+  position: "absolute",
+  top: 0,
+  left: "50%",
+  transform: "translateX(-50%)",
+
+  //zIndex : -1,
+}));
+const StyledImg = styled("img")(() => ({
+  position: "absolute",
+  top: "1%",
+  left: "1%",
+  width: "100px",
+  height: "100px",
+  zIndex: 10000,
+}));
+
+const AccountSetupPage = () => {
+  const [docRef, loading, error] = FirestoreDocumentData();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const navigate = useNavigate();
+
+  // remove next line after test
+  const sleep = (ms) => new Promise((r) => setTimeout(r, ms));
+  const formik = useFormik({
+    initialValues: {
+      headingsFont: "",
+      headingPointSize: "",
+      mainTextFont: "",
+      mainTextPointSize: "",
+      systemName: "",
+      twoFactAuth: false,
+    },
+    validationSchema: accountSetupSchema,
+    onSubmit: async (values) => {
+      setIsSubmitting(true);
+      try {
+        await updateDoc(docRef,{ userPreferences: values });
+        navigate("/");
```

Comment:
This does not seem to work for me for some reason... I wonder if there is something weird going on from a routing perspective